assembly/src/assembler/instruction/procedures.rs

```diff
@@ -58,7 +58,7 @@ impl Assembler {
                         callee: proc.fully_qualified_name().clone(),
                     })
                     .and_then(|module| {
-                        if module.is_kernel() {
+                        if module.unwrap_ast().is_kernel() {
```

Comment:
Are we guaranteed that we get an AST module here?

Reply:
I added a comment as to why I believe this one is OK (please double-check the reasoning):

```rs
// Note: this module is guaranteed to be of AST variant, since we have the
// AST of a procedure contained in it (i.e. `proc`). Hence, it must be that
// the entire module is in AST representation as well.
```